src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,232 @@
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    rpp_hip_math_log(src_f8, dst_f8);
+}
+
+// -------------------- Set 2 - log kernels --------------------
+template <typename T, typename U>
+__global__ void log_1d_hip_tensor(T *srcPtr,
+                                  uint srcStrides,
+                                  U *dstPtr,
+                                  uint dstStrides,
+                                  uint *roiTensor)
+{
+    uint id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8; // width
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;       // batchsize
+
+    uint *roi = &roiTensor[id_z * 2];
+    uint xBegin = roi[0];
+    uint width = roi[1];
+
+    if (id_x >= width)
+        return;
+
+    uint srcIdx = (id_z * srcStrides) + id_x + xBegin;
+    uint dstIdx = (id_z * dstStrides) + id_x;
+
+    d_float8 src_f8, dst_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);
+    log_hip_compute(srcPtr, &src_f8, &dst_f8);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+}
+
+template <typename T, typename U>
+__global__ void log_2d_hip_tensor(T *srcPtr,
+                                  uint2 srcStridesNH,
+                                  U *dstPtr,
+                                  uint2 dstStridesNH,
+                                  uint *roiTensor)
+{
+    uint id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8; // width
+    uint id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;       // height
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;       // batchsize
+
+    uint *roi = &roiTensor[id_z * 4];
+    uint yBegin = roi[0];
+    uint xBegin = roi[1];
+    uint height = roi[2];
+    uint width = roi[3];
+
+    if (id_x >= width || id_y >= height)
+        return;
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + yBegin) * srcStridesNH.y) + id_x + xBegin;
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x;
+
+    d_float8 src_f8, dst_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);
+    log_hip_compute(srcPtr, &src_f8, &dst_f8);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+}
+
+template <typename T, typename U>
+__global__ void log_3d_hip_tensor(T *srcPtr,
+                                  uint2 srcStridesDH,
+                                  U *dstPtr,
+                                  uint2 dstStridesDH,
+                                  uint *roiTensor)
+{
+    uint id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x; // lengthX
+    uint id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y; // lengthY
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z; // lengthZ
+
+    uint *roi = roiTensor;
+    uint zBegin = roi[0];
+    uint yBegin = roi[1];
+    uint xBegin = roi[2];
+    uint lengthZ = roi[3];
+    uint lengthY = roi[4];
```

Comment:
Same comment here. Use a d_uint6 for 3D instead of the uint4 for 2D.
Also stick to one style in your variables - Say consistently use X at the end like beginX, lengthX.